strawberry/relay/types.py

```diff
@@ -375,6 +374,11 @@ def __init_subclass__(cls, **kwargs: Any):
         ]
 
         if len(candidates) == 0:
+            # resolve_id is overwritten, allow unset _id_attr
+            for mro_cls in cls.__mro__:
+                if mro_cls != Node:
+                    if "resolve_id" in cls.__dict__:
+                        return
```

Comment:
nit: compare with `is` and avoid nesting ifs

```suggestion
                if mro_cls is not Node and "resolve_id" in cls.__dict__:
                    return
```

Reply:
done, and edge case solved